src/coreclr/jit/codegenarmarch.cpp

```diff
@@ -362,9 +362,17 @@ void CodeGen::genCodeForTreeNode(GenTree* treeNode)
             genCodeForSelect(treeNode->AsConditional());
             break;
 
+        case GT_CINC:
+            genCodeForCinc(treeNode->AsConditional());
```

Comment:
I think you should just use `GenTreeOp` for `GT_CINC` and make it a normal binary node.

Reply:
It will avoid most of the boilerplate of this PR and be more consistent with the rest of the JIT.

Reply:
Done.